g2utils/um2grb2.py

```diff
@@ -2451,16 +2458,21 @@ def convertFilesInParallel(fnames, ftype):
 
 def _checkInFilesStatus(path, ftype, pfnames):
     
-    global __max_long_fcst_hours__
+    global __start_long_fcst_hour__, __end_long_fcst_hour__
     
     if ftype in ['ana', 'anl']:
         fhrs = ['000'] 
     elif ftype in ['fcst', 'prg']:
+        # calculate start hour of long fcst in multiples of 24. Why?
+        # 00 hr contains from 06 to 24 hours data.
+        # 24 hr contains from 24 to 48 hours data, and so on.
+        start_fcst_hour = (__start_long_fcst_hour__ / 24) * 24
```

Comment:
This is buggy code.
It should be "start_fcst_hour = ((__start_long_fcst_hour__ / 24) - 1) * 24", then only it can produce proper 24th hour file.